### ISSUE TITLE
0.9.5-1 breaks bzip3 decompression

### ISSUE THREAD

**Issue by andreasstieger** [Genivia/ugrep-indexer#9]
With 0.9.4:

````
[    6s] + ./configure --host=x86_64-suse-linux-gnu --build=x86_64-suse-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bzip3
[   12s] checking if bzip3 is wanted... yes
[   12s] checking for bz3_new in -lbzip3... yes
````

With 0.9.5-1:

````
[    2s] [22/173] keeping libbzip3-0-1.4.0-1.1
[    2s] [130/173] keeping bzip3-devel-1.4.0-1.1
[    4s] + ./configure --host=x86_64-suse-linux-gnu --build=x86_64-suse-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bzip3
[    8s] checking if bzip3 is wanted... yes
[    8s] checking for bz3_new in -lbzip3... no
[    8s] optional libbzip3 not found: install if you want ugrep-indexer option -z to index .bz3 files
````

**Comment by barsnick** [comment]
`git bisect` says this was broken by commit https://github.com/Genivia/ugrep-indexer/commit/4e03c7811352b3e70ea71098aab1ee09a36a517c. I cannot tell why though.

**Comment by genivia-inc** [comment]
I don't know why either, at least not yet. I checked the diffs, but the parts that check the presence of the library was not changed, i.e. this line and surrounding lines:
~~~
AC_CHECK_LIB([bzip3], [bz3_new], [bzip3_cv_libbzip3=yes], [bzip3_cv_libbzip3=no])
~~~
I know that libbzip3 was updated recently. But this looks OK:
~~~
nm -g /opt/homebrew/lib/libbzip3.a | fgrep bz3_new
    15:	0000000000000064 T _bz3_new
~~~

**Comment by andreasstieger** [comment]
Move the bzip3 block before the 7z block. In other words, the LZMA flags affect the bzip3 detection.

**Comment by genivia-inc** [comment]
@andreasstieger yep, that's it. I should restore the `CPPFLAGS` and `LDFLAGS` in the 7zip block.

**Comment by andreasstieger** [comment]
Another one for https://github.com/Genivia/ugrep/issues/339 :-)

**Comment by genivia-inc** [comment]
Yeah. But this kind of detection interference is not good.

**Comment by genivia-inc** [comment]
Fixed in update 0.9.5-2.
